syntax = "proto3";
package monitor;

message CommandRequest {
    string command = 1;
}

message MetricValue {
    string name = 1;
    string value = 2;
}

message CommandResponse {
    string timestamp = 1;
    string hostname = 2;
    repeated MetricValue metrics = 3;
}

service MonitorService {
    rpc CommandStream (stream CommandResponse) returns ( CommandRequest);
}
